.github/workflows/helm_pr_deployment_delete.yml

```diff
@@ -0,0 +1,36 @@
+name: Cleanup PR Deployment from Kubernetes
+
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  destroy:
+    if: ${{ github.event.pull_request.head.repo.full_name == 'ls1intum/Artemis' }}
+    runs-on: ubuntu-latest
+    environment: k8s
```

Comment:
This would require the manual approval of you or Matthias to be run. If you forget this, then you'd have an orphaned test server in K8s.

Reply:
That is correct. The thing is: We need this in order to ensure the access credentials to the k8s cluster are not being leaked. If we don't have this, somebody could change the pipeline and extract access to the k8s cluster during deploy and destroy.

At least that is how I understand it. However, I usually don't work with GitHub actions so if you have another way to handle this I'm open to hear it.

Ideally failing pipelines wouldn't be as normalized as they currently are. Then the failing / pending pipeline would serve as a reminder for an orphaned k8s deployment. Another option would be to automatically delete Artemis deployments on k8s after a while. Either way, I think It's still better than the current test server deployments since we don't have to maintain locks on individual servers anymore and can spawn and delete those deployments with ease.

Reply:
I'll mark this as resolved, see my answer to Yanniks feedback